src/datasources/transaction-api/transaction-api.service.ts

```diff
@@ -96,46 +96,49 @@ export class TransactionApi implements ITransactionApi {
     await this.cacheService.deleteByKey(cacheKey);
   }
 
-  async getDataDecoded(data: string, to?: string): Promise<DataDecoded> {
+  async getDataDecoded(args: {
+    data: string;
+    to?: string;
+  }): Promise<DataDecoded> {
```

Comment:
This is a bit excessive imo. Should we now _always_ use an object for 2-3 arguments? (I won't mark other functions likr this.)

Reply:
I've considered not doing this for an object with 2 or 3 arguments. One reason why I decided to push this for these functions is mostly because of what I wrote in the PR description (bullet point 3).

One of the current challenges that we have is that we do not have strong types on some arguments (most of them are `string` and can be used interchangeably).

With this change we would need to specify the name for each argument. I have no strong opinion on this one 🤔 . I'd much prefer something like:

```typescript
foo(a: A, b: B, c: C)
```

Reply:
I've tried to keep the instances where even with 2 or more arguments the types are different.

Examples: 

```typescript
getSafeAppById(chainId: string, id: number): Promise<SafeApp | null>;
```

```typescript
registerDevice(device: Device, safeRegistration: SafeRegistration): Promise<SafeRegistration>;
```

What do you think about these examples?

Reply:
In my opinion, it's OK to use interfaces even for 2-3 argument functions on the repositories/datasources layer as the PR does. These are not directly covered by unit tests, so make it harder to introduce a bug.

Going through the path of implementing something like:
```
foo(a: A, b: B, c: C)
```
If I understand it correctly, we'd need to wrap primitives into something like _DDD value objects_, or even group arguments together into more complex structs that we could reuse. Not sure this is what are you proposing exactly, but depending on how we use it, I think it might be a bit overkill for the current purpose/nature of the project.